offer-manager-app/src/main/groovy/life/qbic/portal/offermanager/offergeneration/QuotationDetails.groovy

```diff
@@ -93,19 +102,43 @@ class QuotationDetails {
     private void generateProductTable(List<OfferItem> items, ProductGroup productGroup, double subTotal) {
         //Create the items in html in the overview table
         //Check if there are ProductItems stored in list
+
+        String productGroupTableId = "${productGroup.toString()}-table"
+
         if (!items) {
-            htmlContent.getElementById("${productGroup.toString()}-table").remove()
+            htmlContent.getElementById(productGroupTableId).remove()
             return
         }
+
         //Clear Template content
         String elementId = generateElementID(0, productGroup)
-        htmlContent.getElementById(elementId).empty()
+        htmlContent.getElementById(productGroupTableId).empty()
 
-        if (isOverflowingPage()) {
-            generateHTMLTableOnNextPage("${productGroup.toString()}-table")
+        //Account for space necessary for Table Title
+        consumedPageSpace += TableElementSpacing.TABLE_TITLE.getTableElementSpacing()
+
+        //Account for Table Header space
+        consumedPageSpace += TableElementSpacing.TABLE_HEADER.getTableElementSpacing()
```

Comment:
It seems like this should be not directly added but also tested such as you did it with the isOverflowingPage() method. Looks redundant to add space for the title and header 2 times

Reply:
but anyhow, I like how easy it is to read your code :)

Reply:
Thanks, I'll make the intend behind it a little bit clearer in the code. 
Adding the space before checking isOverflowingPage ensures that the space is big enough for the first item and the table title and table header. 
This avoids a cutoff between the headers/title and the items